File: python/classes.py
# Classes and OOP in Python

class Animal:
    """Base class for animals."""
    
    def __init__(self, name):
        self.name = name
    
    def speak(self):
        raise NotImplementedError("Subclasses must implement speak()")

class Dog(Animal):
    """A dog that barks."""
    
    def speak(self):
        return f"{self.name} says woof!"

class Cat(Animal):
    """A cat that meows."""
    
    def speak(self):
        return f"{self.name} says meow!"

# Usage
dog = Dog("Rex")
cat = Cat("Whiskers")
print(dog.speak())  # Rex says woof!
print(cat.speak())   # Whiskers says meow!

File: javascript/classes.js
// Classes in JavaScript (ES6)

class Animal {
    constructor(name) {
        this.name = name;
    }
    
    speak() {
        throw new Error("Subclasses must implement speak()");
    }
}

class Dog extends Animal {
    speak() {
        return `${this.name} says woof!`;
    }
}

class Cat extends Animal {
    speak() {
        return `${this.name} says meow!`;
    }
}

// Usage
const dog = new Dog("Rex");
const cat = new Cat("Whiskers");
console.log(dog.speak()); // Rex says woof!
console.log(cat.speak());  // Whiskers says meow!
